shared/runtime/softtimer.h

```diff
@@ -44,6 +44,7 @@ typedef struct _soft_timer_entry_t {
         void (*c_callback)(struct _soft_timer_entry_t *);
         mp_obj_t py_callback;
     };
+    void *context;
```

Comment:
To avoid adding this to all users of this struct it might be possible to instead create a derived struct like this:
```c
typedef struct _soft_timer_entry_extended_t {
    soft_timer_entry_t base;
    void *context;
} soft_timer_entry_extended_t;
```
Then you can pass `&entry->base` to the existing soft-timer functions, but when you get a callback you can cast the `soft_timer_entry_t*` back to a `soft_timer_entry_extended_t*` and access the context.

Reply:
Thanks for the hint. The code is changed now and tested with the SAMD and Renesas port, which are affected by the change.
I could not test with the Pico2, since I have no board yet. But when adding the RP2 commit to the RP2350 PR, it still builds fine.